include/swift/Runtime/HeapObject.h

```diff
@@ -97,66 +97,10 @@ HeapObject *swift_initStaticObject(HeapMetadata const *metadata,
 SWIFT_RUNTIME_EXPORT
 void swift_verifyEndOfLifetime(HeapObject *object);
 
-/// A structure that's two pointers in size.
-///
-/// C functions can use the TwoWordPair::Return type to return a value in
-/// two registers, compatible with Swift's calling convention for tuples
-/// and structs of two word-sized elements.
-template<typename A, typename B>
-struct TwoWordPair {
-  A first;
-  B second;
-  
-  TwoWordPair() = default;
-  TwoWordPair(A first, B second);
-
-  // FIXME: rdar://16257592 arm codegen doesn't call swift_allocBox correctly.
-  // Structs are returned indirectly on these platforms, but we want to return
-  // in registers, so cram the result into an unsigned long long.
-  // Use an enum class with implicit conversions so we don't dirty C callers
-  // too much.
-#if __arm__ || __i386__ || defined(__CYGWIN__) || defined(_MSC_VER)
```

Comment:
One consideration is that this could cause any of these platforms that tries to build without `SWIFT_USE_SWIFTCALL` (e.g. with an unsupported Clang) to silently break. Is there any reason that `SWIFT_USE_SWIFTCALL` can't be required now? Otherwise it might be worth adding back an `#if` block for these platforms with an `#error` that the Swift calling convention must be enabled.

Reply:
Good point. If we start requiring `SWIFT_USE_SWIFTCALL` then we ought to remove its other uses in `Config.h` and add the `#error` check there.

I see at least one reference to the C compiler being MSVC. I assume it has no swiftcall support. Do we actually support MSVC? Is anyone using it? 

Formally requiring recent clang would be worth a heads-up notice or a discussion on swift-evolution.

Reply:
Also, I see that the default definition of `SWIFT_USE_SWIFTCALL` in `Config.h` still disables it. Are you building locally with it enabled?

Reply:
Yes; I've submitted a [PR for Clang](https://reviews.llvm.org/D40929) that means we can enable it on Windows when using Clang.

As for MSVC: I know @hughbe was doing a lot of work around getting the compiler building under MSVC. However, if I'm not mistaken, `SWIFT_USE_SWIFTCALL` and its descendants are only used in the stdlib/runtime, which is normally built using the just-built Clang anyway. The requirement then is a little less strict: the stdlib/runtime must be built with a recent Clang - and I think that was already a requirement.

Config.h is pulled in from a few non-runtime sources, so just erroring out within that file might cause issues for people trying to build swiftc with MSVC etc., but I don't see any issue with moving the calling convention parts into a separate file.

Reply:
I think that distinction will work. Probably looks something like this:
* Remove `SWIFT_USE_SWIFTCALL` from Config.h. (In the interim add some `#if !SWIFT_USE_SWIFTCALL #error "SWIFT_USE_SWIFTCALL=0 is not supported"` to catch anyone switching it externally.)
* Use `__has_attribute(swiftcall)` to define the macros formerly controlled by `SWIFT_USE_SWIFTCALL`. Or possibly define them unconditionally, if they are never used in code that is included in the compiler's build.
* Add `__has_attribute(swiftcall)` enforcement somewhere in runtime-only code. I don't know of any established place for such a thing.